react-cors-spa/src/EditProductDialog.jsx

```diff
@@ -0,0 +1,175 @@
+import { Fragment, useEffect, useState } from "react";
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  TextField,
+  DialogActions,
+  Button,
+  Box,
+  Typography,
+} from "@mui/material";
+import { useForm } from "react-hook-form";
+import CloseOutlinedIcon from "@mui/icons-material/CloseOutlined";
+import { useQueryClient, useMutation } from "@tanstack/react-query";
+import modifyProduct from "./services/ProductService";
+import "./EditProductDialog.css";
+import SelectCategory from "./SelectCategory";
+
+function EditProductDialog({
+  toggleEditDialog,
+  editDialogOpen,
+  selectedProduct,
+  refetch,
+}) {
+  const { id, name, description, sku, size, category_id } = selectedProduct;
+
+  const [sure, setSure] = useState(false);
+
+  const queryClient = useQueryClient();
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    reset,
+    formState: { errors },
+  } = useForm();
+
+  const submitProductMutation = useMutation({
+    mutationFn: (productData) => {
+      return modifyProduct(productData);
+    },
+    onSuccess: () => queryClient.invalidateQueries({ queryKey: ["products"] }),
+  });
+
+  function Submit(productData) {
+    refetch();
+    submitProductMutation.mutate({
+      product_id: Number(id),
+      ...productData,
+      category_id: productData.category_id,
+    });
+    reset();
+    toggleEditDialog();
+    setSure(false);
+  }
+
+  function CloseDialog() {
+    reset();
+    toggleEditDialog();
+    setSure(false);
+  }
+
+  useEffect(() => {
+    const product = {
+      name,
+      description,
+      sku,
+      size,
+      category_id,
+    };
+    reset(product);
+  }, [reset, name, description, sku, size, category_id]);
+
+  return (
```

Comment:
Typically I would say that you shouldn't separate the add and edit into two different components because you end up repeating a lot of code. Let's not worry about it here, but think how you might be able to serve both add and edit functionality from one component.